pypesto/ensemble/covariance_analysis.py

```diff
@@ -0,0 +1,318 @@
+        return only separable directions according to lower_cutoff
+
+    Returns
+    =======
+    eigen_values:
+        Eigenvalues of the covariance matrix
+
+    eigen_vectors:
+        eigenvectors of the covariance matrix
+    """
+    covariance = get_covariance_matrix_parameters(ens)
+    return get_spectral_decomposition_lowlevel(
+        covariance, normalize, lower_absolute_cutoff, lower_relative_cutoff,
+        inv_upper_absolute_cutoff, inv_upper_relative_cutoff,
+        only_identifiable_directions, only_separable_directions)
+
+
+def get_spectral_decomposition_predictions(
+        ens: Ensemble,
+        normalize: bool = False,
+        lower_absolute_cutoff: float = None,
+        lower_relative_cutoff: float = None,
+        inv_upper_absolute_cutoff: float = None,
+        inv_upper_relative_cutoff: float = None,
+        only_identifiable_directions: bool = False,
+        only_separable_directions: bool = False):
+    """
+    Compute the spectral docmposition of ensemble predictions.
+
+    Parameters
+    ==========
+    ens:
+        Ensemble object containing a set of parameter vectors and a set of
+        predictions or EnsemblePrediction object containing only predictions
+
+    normalize:
+        flag indicating whether the returned Eigenvalues should be normalized
+        with respect to the largest Eigenvalue
+
+    lower_absolute_cutoff:
+        Consider only eigenvalues of the covariance matrix above this cutoff
+        (only applied when only_separable_directions is True)
+
+    lower_relative_cutoff:
+        Consider only eigenvalues of the covariance matrix above this cutoff,
+        when rescaled with the largest eigenvalue
+        (only applied when only_separable_directions is True)
+
+    inv_upper_absolute_cutoff:
+        Consider only low eigenvalues of the covariance matrix with inverses
+        above of this cutoff
+        (only applied when only_identifiable_directions is True)
+
+    inv_upper_relative_cutoff:
+        Consider only low eigenvalues of the covariance matrix when rescaled
+        with the largest eigenvalue with inverses above of this cutoff
+        (only applied when only_identifiable_directions is True)
+
+    only_identifiable_directions:
+        return only identifiable directions according to inv_upper_cutoff
+
+    only_separable_directions:
+        return only separable directions according to lower_cutoff
+
+    Returns
+    =======
+    eigen_values:
+        Eigenvalues of the covariance matrix
+
+    eigen_vectors:
+        eigenvectors of the covariance matrix
+    """
+    covariance = get_covariance_matrix_predictions(ens)
+    return get_spectral_decomposition_lowlevel(
+        covariance, normalize, lower_absolute_cutoff, lower_relative_cutoff,
+        inv_upper_absolute_cutoff, inv_upper_relative_cutoff,
+        only_identifiable_directions, only_separable_directions)
+
+
+def get_covariance_matrix_lowlevel(dataset):
```

Comment:
Is it expected that this function may get more complicated? If not, could remove it?

Reply:
Maybe in some way... On might want to choose whether we want to use the algorithms from scipy or numpy... but not in the next future. You're right, will remove it...